### PR TITLE
add `string[]` to query param types

### PR DESCRIPTION
**Pull request by sachinraja**
A query parameter can also be a string array when you're reading it and creating it.

### CODE REVIEW

File: src/core.ts [hunk 1]
```diff
@@ -153,7 +156,7 @@ declare module "next/router" {
     [K in Route as K["pathname"]]: Exclude<K["query"], undefined>;
   };
 
-  type RouteOrQuery = Route | { query: { [key: string]: string | undefined } };
+  type RouteOrQuery = Route | { query?: UnknownQueryParams };
```
Comment (sachinraja) [inline src/core.ts:159]: Not sure if `query` was intentionally not optional here, but I changed it to align with the `next/link` one.

Comment (tatethurston) [reply src/core.ts:159]: It was, my thinking was that if you don't provide Route, you should provide a query, thus making `{}` an invalid argument. I assumed `{}` was rare enough that it was most likely an accident, and users could cast if it was intentional.

I don't feel particularly strongly about this stance thought

File: src/__snapshots__/core.test.ts.snap [hunk 2]
```diff
@@ -8,12 +8,15 @@ Array [
 // Run \`npx nextjs-routes\` to regenerate this file.
 /* eslint-disable */
 
+// prettier-ignore
+type UnknownQueryParams = { [key: string]: string | string[] | undefined }
```
Comment (tatethurston) [inline src/__snapshots__/core.test.ts.snap:12]: Could you move this into the declare module nextjs-routes below and then import it at the call sites? That way it's not globally available.

**Comment by tatethurston** [comment]
Good catch, thanks for this

File: src/core.ts [hunk 3]
```diff
@@ -84,14 +84,13 @@ function getQueryInterface(
 }
 
 function generate(routes: Route[]): string {
+  const unknownQueryParamsType =
+    "{ [key: string]: string | string[] | undefined }";
```
Comment (tatethurston) [inline src/core.ts:88]: 👍 This is a nice way to control the exported namespaces while still cutting down on duplication

**Review by tatethurston — APPROVED** [review]

**Pull request merged**